### PR TITLE
Add a github wrokflow for tests

### PR DESCRIPTION
**Pull request by albacoretuna**

### CODE REVIEW

File: backend/src/graphql/resolvers-types.ts [hunk 1]
```diff
@@ -2,12 +2,9 @@ import { GraphQLResolveInfo, GraphQLScalarType, GraphQLScalarTypeConfig } from '
 import { ApolloContext } from '../server/apolloContext';
 export type Maybe<T> = T | null;
 export type Exact<T extends { [key: string]: unknown }> = { [K in keyof T]: T[K] };
-export type MakeOptional<T, K extends keyof T> = Omit<T, K> &
```
Comment (pouriaMaleki) [inline backend/src/graphql/resolvers-types.ts:5]: This change is not related to this merge request. 

(Why there is formatting change? 🤔  `./backend/codegen.yml` should run the prettier automatically after each update)

File: .github/workflows/test.js.yml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Use Node.js ${{ matrix.node-version }}
+        working-directory: ./backend
```
Comment (pouriaMaleki) [inline .github/workflows/test.js.yml:23]: Can you please add the frontend type check too?

(`next dev` doesn't check types but `next build` does and it makes it possible to miss something during the development.)

File: backend/src/graphql/resolvers-types.ts (hunk 1 above)
Comment (albacoretuna) [reply backend/src/graphql/resolvers-types.ts:5]: Yeah, I've done too many tests on this branch. I won't merge it. I'll just copy the final action file to a new branch :)

File: .github/workflows/test.js.yml (hunk 2 above)
Comment (albacoretuna) [reply .github/workflows/test.js.yml:23]: Sure, after backend is working, I'll add frontend.

**Comment by albacoretuna** [comment]
I close this and merge #29
